R/messages.R

```diff
@@ -110,3 +110,85 @@ setvar <- function(core, dates, var, values, unit)
 
     invisible(NULL)
 }
+
+
+#' Fetch all the available Hector variables
+#'
+#' This function is similar to fetchvars, except it fetches all available
+#' Hector variables, including exogenous inputs (e.g., emissions & pre-industrial
+#' concentrations), parameters (e.g., alpha, beta, ECS), and output
+#' variables (e.g., concentrations & forcings).
+#'
+#' @param core Hector core object
+#' @param dates Vector of dates, optional
+#' @param scenario Scenario name, optional str
+#' @param outpath Absolute path of the output csv. Output will only be written to
+#' csv if this is given
+#' @return vars_all    Dataframe containing all Hector variables
+#' @family main user interface functions
+#' @export
+fetchvars_all <- function(core, dates=NULL, scenario=NULL, outpath=NULL) {
+
+    if (is.null(scenario)) {
+        scenario <- core$name
+    }
+
+    strt <- startdate(core)
+    end <- getdate(core)
+
+    if (is.null(dates)) {
+        dates <- strt:end
+    } else {
+        valid <- dates >= strt & dates <= end
+        dates <- dates[valid]
+    }
```

Comment:
If you use `fetchvars` (as indicated below), most of these lines should no longer be necessary.